Tools/LambdaTestTool/src/Amazon.Lambda.TestTool/Utils.cs

```diff
@@ -241,5 +242,66 @@ public static string SearchLatestCompilationDirectory(string debugDirectory)
 
             return depsFile[0].Directory.FullName;            
         }
+
+        /// <summary>
+        /// Returns the Lambda assembly file path that will debugged by the test tool.
+        /// It returns an empty string if no Lambda assembly path is found.
+        /// </summary>
+        /// <param name="debugDirectory">This points to the .../bin/{CONFIGURATION}/{TARGET_FRAMEWORK} directory</param>
+        public static string FindLambdaAssemblyPath(string debugDirectory)
+        {
+            var depsFiles = Directory.GetFiles(debugDirectory, "*.deps.json");
+            if (!depsFiles.Any())
+            {
+                return string.Empty;
+            }
+
+            if (depsFiles.Length == 1)
+            {
+                var depsFilePath = depsFiles.First();
+                var lambdaAssemblyPath = depsFilePath.Substring(0, depsFilePath.Length - ".deps.json".Length) + ".dll";
+                return lambdaAssemblyPath;
+            }
+
+            var dependencies = new HashSet<string>();
+            foreach (var depsFilePath in depsFiles)
+            {
+                var depsFileContent = File.ReadAllText(depsFilePath);
+                ExtractDependenciesFromDepsJson(JsonDocument.Parse(depsFileContent).RootElement, dependencies);
+            }
+
+            foreach (var depsFilePath in depsFiles)
+            {
+                var depsFileName = Path.GetFileName(depsFilePath);
+                var projectName = depsFileName.Substring(0, depsFileName.Length - ".deps.json".Length);
+                if (!dependencies.Contains(projectName))
+                {
+                    var lambdaAssemblyPath = depsFilePath.Substring(0, depsFilePath.Length - ".deps.json".Length) + ".dll";
+                    return lambdaAssemblyPath;
+                }
+            }
+
+            return string.Empty;
+        }
+
+        private static void ExtractDependenciesFromDepsJson(JsonElement node, HashSet<string> dependencies)
+        {
+            if (node.ValueKind != JsonValueKind.Object)
+                return;
+
+            if (node.TryGetProperty("dependencies", out var depenciesBlob)) 
+            {
+                foreach (var dependency in depenciesBlob.EnumerateObject())
+                {
+                    dependencies.Add(dependency.Name);
```

Comment:
Since the same dependency could be used across multiple assemblies you should check to see if `dependency.Name` has already been added to `dependencies`.

Reply:
The dependencies collection is already a Hashset so we don't need to check for duplicates.

Reply:
My mistake. I was thinking the `Add` method would throw an exception if attempting to add a duplicate. I see now that the `Add` returns back a boolean whether the item was added.